src/backend/distributed/commands/statistics.c

```diff
@@ -445,19 +446,30 @@ GetExplicitStatisticsCommandList(Oid relationId)
 
 		if (alterStatisticsOwnerCommand != NULL)
 		{
-			alterStatisticsCommandList =
-				lappend(alterStatisticsCommandList,
+			explicitStatisticsCommandList =
+				lappend(explicitStatisticsCommandList,
 						makeTableDDLCommandString(alterStatisticsOwnerCommand));
 		}
 	}
 
+	/* find indexes on current relation, in case of modified stats target */
+	Relation relation = relation_open(relationId, AccessShareLock);
+	List *indexOidList = RelationGetIndexList(relation);
+	relation_close(relation, AccessShareLock);
+
+	Oid indexOid = InvalidOid;
+	foreach_oid(indexOid, indexOidList)
+	{
+		List *alterIndexStatisticsCommands = GetAlterIndexStatisticsCommands(indexOid);
+
+		explicitStatisticsCommandList =
+			list_concat(explicitStatisticsCommandList, alterIndexStatisticsCommands);
+	}
```

Comment:
at least we can move this block into a helper function and the above `statisticsIdList` loop into another. Then we can also comment what kind of statistic object we are catching in those two functions to further readers (including me :))